File: app/src/main/java/android/mahendra/attendancemanager/viewmodels/PeriodListViewModel.java
package android.mahendra.attendancemanager.viewmodels;

import android.app.Application;
import android.mahendra.attendancemanager.models.Period;
import android.mahendra.attendancemanager.models.Subject;
import android.mahendra.attendancemanager.repositories.PeriodRepository;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

public class PeriodListViewModel extends AndroidViewModel {
    private PeriodRepository mPeriodRepository;
    private LiveData<List<Period>> mAllPeriods;

    public PeriodListViewModel(@NonNull Application application) {
        super(application);
        mPeriodRepository = new PeriodRepository(application);
        mAllPeriods = mPeriodRepository.getAllPeriods();
    }

    public void insert(Period period) {
        mPeriodRepository.insert(period);
    }

    public void update(Period period) {
        mPeriodRepository.update(period);
    }

    public LiveData<List<Period>> getAllPeriods() {
        return mAllPeriods;
    }

    public LiveData<List<Period>> getAllPeriodsOn(int weekDay) {
        return mPeriodRepository.getAllPeriodsOn(weekDay);
    }

    public LiveData<List<Subject>> getAllSubjectsOn(int weekDay) {
        return mPeriodRepository.getAllSubjectsOn(weekDay);
    }
}

File: app/src/main/java/android/mahendra/attendancemanager/viewmodels/PeriodListViewModel.kt
package android.mahendra.attendancemanager.viewmodels

import android.app.Application
import android.mahendra.attendancemanager.models.Period
import android.mahendra.attendancemanager.models.Subject
import android.mahendra.attendancemanager.repositories.PeriodRepository

import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.launch

import kotlin.collections.List

class PeriodListViewModel(application: Application) : AndroidViewModel(application) {
    private val periodRepository: PeriodRepository = PeriodRepository(application)
    private val allPeriods: LiveData<List<Period>> = periodRepository.mAllPeriods

    fun insert(period: Period) = viewModelScope.launch {
        periodRepository.insert(period)
    }

    fun update(period: Period) = viewModelScope.launch {
        periodRepository.update(period)
    }

    fun getAllPeriods(): LiveData<List<Period>> {
        return allPeriods
    }

    fun getAllPeriodsOn(weekDay: Int): LiveData<List<Period>> {
        return periodRepository.getAllPeriodsOn(weekDay)
    }

    fun getAllSubjectsOn(weekDay: Int): LiveData<List<Subject>> {
        return periodRepository.getAllSubjectsOn(weekDay)
    }

    fun deletePeriod(periodNumber: Int, weekDay: Int) = viewModelScope.launch {
        periodRepository.deletePeriod(periodNumber, weekDay)
    }
}
